src/components/FileDownloaderLink.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useRef, useEffect, AnchorHTMLAttributes } from 'react'
+
+interface FileSaverLinkProps {
+  data?: string | (() => string)
+  options?: BlobPropertyBag
+  filename: string
+}
+
+export const FileDownloaderLink: React.FC<FileSaverLinkProps & AnchorHTMLAttributes<HTMLAnchorElement>> = ({
+  data,
+  options,
+  filename,
+  ...anchorProps
+}) => {
+  const lastObjectURL = useRef('')
+  const lastDataUsed = useRef('')
+
+  useEffect(
+    () => (): void => {
+      // on unmount release ObjectURL if any
+      if (lastObjectURL.current) URL.revokeObjectURL(lastObjectURL.current)
+    },
+    [],
+  )
+
+  // only create download onClick because
+  // creating and retaining BLob and ObjectURL can be expensive on each render
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
+    // no data
+    if (!data) {
+      // prevent invalid file download
+      e.preventDefault()
+      return
+    }
+    const csvData = typeof data === 'string' ? data : data()
+    // data didn't change, reusing already set ObjectURL
+    if (lastDataUsed.current === csvData) return
+    lastDataUsed.current = csvData
+
+    // new data, need new ObjectURL
+    // release old if was set previously
+    if (lastObjectURL.current) URL.revokeObjectURL(lastObjectURL.current)
+
+    const blob = new Blob([csvData], options)
```

Comment:
Wouldn't make sense to use a more stream approach?

We usually represent lists of something, so probably make sense, so we don't run into issues in bigger files?

Reply:
How would you use a stream here?